### ISSUE TITLE
False positive detection on function that is not a rule function

### ISSUE THREAD

**Issue by bradzacher** [eslint-community/eslint-plugin-eslint-plugin#455]
Version: 5.5.0

Relevant Code: https://github.com/typescript-eslint/typescript-eslint/blob/fde44275117b3a7bbdcaaf267c73986b6637c5f9/packages/eslint-plugin/tests/rules/prefer-optional-chain/base-cases.ts#L232-L283

Output:
```
/Users/bjz/git/typescript-eslint/packages/eslint-plugin/tests/rules/prefer-optional-chain/base-cases.ts
  232:43  error  `meta.messages` must contain at least one violation message                    eslint-plugin/prefer-message-ids
  232:43  error  Rules should be declared using the object style                                eslint-plugin/prefer-object-rule
  232:43  error  `meta.type` is required (must be either `problem`, `suggestion`, or `layout`)  eslint-plugin/require-meta-type
  232:43  error  `meta.schema` is required (use [] if rule has no schema)                       eslint-plugin/require-meta-schema
```

**Comment by bradzacher** [comment]
Also finding even more across the repo on the same commit as above:
```
      /Users/bjz/git/typescript-eslint/packages/typescript-estree/src/create-program/createIsolatedProgram.ts
        14:1  error  `meta.messages` must contain at least one violation message                    eslint-plugin/prefer-message-ids
        14:1  error  Rules should be declared using the object style                                eslint-plugin/prefer-object-rule
        14:1  error  `meta.type` is required (must be either `problem`, `suggestion`, or `layout`)  eslint-plugin/require-meta-type
        14:1  error  `meta.schema` is required (use [] if rule has no schema)                       eslint-plugin/require-meta-schema
      
      /Users/bjz/git/typescript-eslint/packages/typescript-estree/src/create-program/createSourceFile.ts
        32:1  error  `meta.messages` must contain at least one violation message                    eslint-plugin/prefer-message-ids
        32:1  error  Rules should be declared using the object style                                eslint-plugin/prefer-object-rule
        32:1  error  `meta.type` is required (must be either `problem`, `suggestion`, or `layout`)  eslint-plugin/require-meta-type
        32:1  error  `meta.schema` is required (use [] if rule has no schema)                       eslint-plugin/require-meta-schema
      
      /Users/bjz/git/typescript-eslint/packages/typescript-estree/src/create-program/shared.ts
        40:1  error  `meta.messages` must contain at least one violation message                    eslint-plugin/prefer-message-ids
        40:1  error  Rules should be declared using the object style                                eslint-plugin/prefer-object-rule
        40:1  error  `meta.type` is required (must be either `problem`, `suggestion`, or `layout`)  eslint-plugin/require-meta-type
        40:1  error  `meta.schema` is required (use [] if rule has no schema)                       eslint-plugin/require-meta-schema
```
